### ISSUE TITLE
beacon crashes when adding item to drops

### ISSUE THREAD

**Issue by typicalpanda1772** [thommcgrath/Beacon#198]
beacon on windows 10 crashed whenever I try to add an item to a drop so if I click the plus to search for an item after I type 3 letters it freezes says unresponsive and crashes the whole application

**Comment by typicalpanda1772** [comment]
I have deleted beacon and redownloaded it and it didn't fix it...

**Comment by thommcgrath** [comment]
I’ve heard of this happening in very rare cases, but I don’t have any solid advice for you. My guess would be something to do with database access being restricted, likely due to an over aggressive antivirus. But that doesn’t make a ton of sense because it would happen when just listing the engrams, not just searching.

Normally I’d say updating Beacon would be the solution because the engram search process was updated a while back to prevent the possibility of locking up like you’re seeing. But you said you redownloaded it, so isn’t that either. Just to be certain though, check your Home tab in Beacon. Under the title and logo is the version. It should be 1.2.5.

So... I don’t know. Does it matter which 3 letters you type? I’m guessing no, but it’s worth asking.

**Comment by typicalpanda1772** [comment]
doesn't matter what 3 letters and it's ver 1.2.5 as well as I turned off my anti-virus and didn't fix/change anything

**Comment by typicalpanda1772** [comment]
so as an example if I type metal it will freeze at met or if I type element it will freeze at ele it will freeze  then say unresponsive and then 3 seconds later it crashes

**Comment by typicalpanda1772** [comment]
ohh and idk if this helps but the last code I got from beacon is broken and only works with blue drop nothing else when it should be changing everything/all the drops

**Comment by thommcgrath** [comment]
Regarding your last comment, no that’s completely unrelated. You either updated your server incorrectly (such as using Nitrado’s “extra game ini” or “engine settings” fields) or you didn’t customize all the drops spawned by the map you’re using.

The freezing is so strange to me. There is no code that triggers on the third character. It’s all based on time. So the search process doesn’t start until it thinks you’re done typing. But 3 characters? No, nothing like that. So I’m really scratching my head. Unfortunately, I don’t know if we’ll be able to figure it out.

**Comment by typicalpanda1772** [comment]
ok, I tried to add an item while monitoring task manager and my CPU, disk/SSD and GPU spikes in usage right before I crash while my ram is unaffected I have a good computer to 64 GB of ram i7 8700k intel CPU gtx 1080ti 2x 850 Samsung Evos 500gb as well as a Seagate Baracuda 1 TB so I don't think it's a hardware side dunno if any of this helps

**Comment by thommcgrath** [comment]
The GPU is unusual, but the other two make sense. If the disk access is blocked for some reason, when Beacon tries to read from the database, the CPU will be blocked waiting for the disk. So both will appear at 100%. But Beacon does absolutely nothing with the GPU, so I can’t explain that one.

Let’s try something. Go to Beacon’s help menu, choose Open Data Folder, edit Beacon, and delete the Library.sqlite and any other Library files. You might see two others such as Library.sqlite-wal. Don’t delete anything else, just the files that start with Library.

Relaunch Beacon. It should recreate the Library. Maybe there’s just something wrong with the database and rebuilding it may solve it? I have my doubts, but it’s worth a shot.

**Comment by typicalpanda1772** [comment]
didn't fix it all it did was let me type 4 characters instead of 3 before crashing

**Comment by thommcgrath** [comment]
Here’s a crazy thought. Any chance you’re running at an uneven scaling size, such as 125% or 150%? Since you mentioned the GPU, *maybe* it’s the tag field trying repeatedly to size to a whole number that can’t be done. It’s definitely a long shot, but I’m grasping st straws.

**Comment by typicalpanda1772** [comment]
yes I am using an uneven display size but have had it like that for over a year and it didn't cause problems before

**Comment by typicalpanda1772** [comment]
it is 115% for my scaling size
if that helps

**Comment by thommcgrath** [comment]
But the tag field resizing is a pretty new thing. I’ll set my display to 115% and if I’m lucky, it’ll happen for me too. If I can trigger the behavior, I can fix it.

**Comment by typicalpanda1772** [comment]
okay

**Comment by thommcgrath** [comment]
So at 115% I was not able to reproduce the problem. However, I did have Beacon just blink out of existence while rapidly typing and deleting. This is something I could not reproduce at 100% scaling.

It would be worth seeing if your freeze happens at 100%, but at this point I'm thinking the scaling is unrelated. So we're probably back to the drawing board.

**Comment by typicalpanda1772** [comment]
well i'll try with 100% scaling to see if it fixes

**Comment by typicalpanda1772** [comment]
i switched to 100% scaling and it seems to be fixed

**Comment by thommcgrath** [comment]
Hmm. Ok. I'll keep investigating along those lines then.

**Comment by typicalpanda1772** [comment]
okay thanks for the help and effort i'll just need to switch to 100% when i'm using it
